subprojects/build-operations/src/main/java/org/gradle/trace/buildops/BuildOperationTrace.java

```diff
@@ -0,0 +1,50 @@
+package org.gradle.trace.buildops;
+
+import org.gradle.api.internal.GradleInternal;
+import org.gradle.initialization.BuildRequestMetaData;
+import org.gradle.internal.operations.*;
+import org.gradle.trace.stream.AsyncWriter;
+
+import java.io.File;
+
+@SuppressWarnings("unused")
+public class BuildOperationTrace {
+    public static void start(GradleInternal gradle, File outFile) {
+        BuildOperationListenerManager manager = gradle.getServices().get(BuildOperationListenerManager.class);
```

Comment:
❌ I don't think this will work in all supported Gradle versions. Could we reuse the logic we have in chrome-trace here?

Reply:
At this stage, we're really only interested in measuring this for recent versions of Gradle, so I don't want to put any real effort into making this work for older versions. I did have a quick look at reusing the logic but didn't want the overhead of the adapting layers. But we can if you think we should. For now, I'd rather we just fail if configuration time instrumentation is requested for a Gradle version that it doesn't work with. We can support older versions later if someone needs it.

We also have a similar question with Java version. Some of the instrumentation code is Java 8 but the older versions will work even back to Java 6. I think it's probably ok to just work with Java 8 for now and fail if people try to measure using something older.